engine-precompiles/src/native.rs

```diff
@@ -220,6 +223,56 @@ fn get_nep141_from_erc20(erc20_token: &[u8]) -> AccountId {
     .unwrap()
 }
 
+#[cfg(feature = "contract")]
+fn get_native_erc20_state() -> NativeErc20EngineState {
```

Comment:
We're going to need to work on this. In order for the standalone engine to be able to execute all Aurora transactions exactly the same as on NEAR we cannot have some functions that only with the the NEAR Runtime.

The way we have been doing this for other precompiles is to pass any state needed to the precompile at the time the Engine object is created. That approach may not quite work here, we may need to pass the full `IO` object instead, but I'll need to review more carefully to see.

Reply:
@birchmd hmm, could you please give more details about this. As I see we already have access to the `IO` here https://github.com/aurora-is-near/aurora-engine/blob/bridge-native-aurora-tokens/engine-precompiles/src/native.rs#L215, so I used this code as an example. I don't fully get your point.

Reply:
Hm, I see. I thought we had eliminated all the `contract` feature gates when we did the standalone engine work previously. I guess I must have missed this one. If all this native precompile stuff is already gated behind the `contract` feature then following the same pattern here is ok for now just so that we don't block this feature. But please add a TODO and create a corresponding issue assigned to me so that I can sort this out after your PR is merged.